javascript/javascript3/week3/shopping.js

```diff
@@ -0,0 +1,128 @@
+// sometimes it showing 403 failed loading error
+const user1 = fetch(
+  "https://api.github.com/search/repositories?q=user:benna100"
+)
+  .then((response) => response.json())
+  .then((bennaData) => {
+    return `FullName : ${bennaData.items[1].full_name}<br>
+    url : ${bennaData.items[1].owner.url}<br>
+    Owner : ${bennaData.items[1].owner.login}`;
+  });
+
+const user2 = fetch(
+  "https://api.github.com/search/repositories?q=user:Mamidi001"
+)
+  .then((response) => response.json())
+  .then((mamidiData) => {
+    return `FullName : ${mamidiData.items[1].full_name}<br>
+    url : ${mamidiData.items[1].owner.url}<br>
+    Owner : ${mamidiData.items[1].owner.login}`;
+  });
+
+const user3 = fetch(
+  "https://api.github.com/search/repositories?q=user:amritpandey"
+)
+  .then((response) => response.json())
+  .then((amritData) => {
+    return `FullName : ${amritData.items[3].full_name}<br>
+    url : ${amritData.items[1].owner.url}<br>
+    Owner : ${amritData.items[1].owner.login}`;
+  });
+
+let promiseArray = [user1, user2, user3];
+Promise.all(promiseArray).then((result) => {
+  console.log(result);
+  const ul = document.createElement("ul");
+  for (let i = 0; i < result.length; i++) {
+    const li = document.createElement("li");
+    li.innerHTML = result[i];
+    ul.appendChild(li);
+    document.getElementById("userUrl").appendChild(ul);
+  }
+});
```

Comment:
You are not rendering all the repositories of the user,  just one each and you are not really utilizing the promise.all here. 
The task is "Fetch all the 3 classmates repositories at the same time using Promise.all. "

To sum up, I would advice to:
Make and array of user names and map through the array calling a function that fetches from the github url and returns a promise. 
Do a promise.all on that new mapped array and use the data accordingly: sort through the nested object and make sure to render all (or at least several) user repositories in your chose format. 
For the user data rendering, you can make it a function and just call on every nested part of the object that you will loop through. 

There is good logic and good base here, just some improvements to make sure the task is done and there is no redundant repetitive code 💪🏽